circuit_maintenance_parser/parsers/att.py

```diff
@@ -0,0 +1,114 @@
+"""ATT Parser."""
+
+import logging
+import re
+import string
+from typing import Dict, List
+
+import dateutil
+from bs4.element import ResultSet  # type: ignore
+from circuit_maintenance_parser.parser import CircuitImpact, Html, Impact, Status, Xlsx
+
+
+logger = logging.getLogger(__name__)
+
+RE_EVENT = re.compile(
+    r"Event ID: (.*)[ \n]"
+    r"Customer Impact Description: (.*)[ \n]"
+    r"Summary: (.*)[ \n]"
+    r"Description: (.*)[ \n]"
+    r"Business Risk: (.*)"
+)
+RE_MAINTENANCE_WINDOW_GMT = re.compile(r"Start Time: (.* GMT).*End Time: (.* GMT)")
+RE_MAINTENANCE_WINDOW_NO_TIMEZONE = re.compile(r"Start Time: (.*)[ \n]End Time: (.*)")
+
+
+class XlsxParserATT1(Xlsx):
+    """Xlsx Parser for ATT file attachments."""
+
+    @staticmethod
+    def parse_xlsx(records: List[Dict]) -> List[Dict]:
+        """Parses ATT xlsx attachments."""
+        impact = Impact.OUTAGE
+        account_name, circuit_id_key = get_account_and_circuit_id_key(records[0])
+        circuit_ids = [r[circuit_id_key] for r in records]
+        if "Customer" in records[0]:
+            circuit_ids = [normalize_lec_circuit_id(cid) for cid in circuit_ids]
+        circuits = [CircuitImpact(impact=impact, circuit_id=cid) for cid in circuit_ids]
+        data = [
+            {
+                "account": account_name,
+                "circuits": circuits,
+            }
+        ]
+        return data
+
+
+class HtmlParserATT1(Html):
+    """Notifications Parser for ATT notifications."""
+
+    def parse_html(self, soup):
+        """Parse ATT HTML notification."""
+        logger.debug("Parsing ATT HTML notification.")
+        data = self.parse_p_tags(soup)
+        data["start"] = self.dt2ts(data["start"])
+        data["end"] = self.dt2ts(data["end"])
+        data["status"] = Status.CONFIRMED
+        return [data]
+
+    @staticmethod
+    def parse_p_tags(soup: ResultSet) -> Dict:
+        """Parse <p> tags in HTML."""
+        data = {}
+        p_tags = soup.find_all("p")
+
+        for tag in p_tags:
+            text = tag.text.strip()
+            text = remove_unprintable(text)
```

Comment:
personally, when reusing varialbe names, I read better `text = remove_unprintable(tag.text.strip())`

Reply:
but this is my personal preference